File: app-miyou/src/main/java/com/miyou/app/infrastructure/monitoring/repository/SpringDataMetricsRollupRepository.java
package com.miyou.app.infrastructure.monitoring.repository;

import java.time.Instant;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.monitoring.document.MetricsRollupDocument;
import reactor.core.publisher.Flux;

public interface SpringDataMetricsRollupRepository
	extends
		ReactiveMongoRepository<MetricsRollupDocument, String> {

	Flux<MetricsRollupDocument> findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
		String granularity,
		Instant startTime,
		Instant endTime);
}

File: infrastructure/src/main/kotlin/com/miyou/app/infrastructure/monitoring/repository/SpringDataMetricsRollupRepository.kt
package com.miyou.app.infrastructure.monitoring.repository

import com.miyou.app.infrastructure.monitoring.document.MetricsRollupDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux
import java.time.Instant

interface SpringDataMetricsRollupRepository : ReactiveMongoRepository<MetricsRollupDocument, String> {
    fun findByGranularityAndBucketStartBetweenOrderByBucketStartAsc(
        granularity: String,
        startTime: Instant,
        endTime: Instant,
    ): Flux<MetricsRollupDocument>
}
